cwe_checker_rs/src/intermediate_representation/term.rs

```diff
@@ -68,6 +68,33 @@ pub enum Def {
     Assign { var: Variable, value: Expression },
 }
 
+impl Term<Def> {
+    /// This function checks whether the following instruction
+    /// is a zero extension of the currently overwritten sub register and
+    /// if so, returns its TID
+    pub fn check_for_zero_extension(&self, output_name: String) -> Option<Tid> {
+        match &self.term {
+            Def::Assign { var, value } => {
+                if output_name == var.name {
+                    match value {
+                        Expression::Cast { op, arg, .. } => match op {
+                            CastOpType::IntZExt => match **arg {
+                                Expression::Var(_) => Some(self.tid.clone()),
```

Comment:
I only check if the output register of the zero extension is the correct one. Overwriting the full register with a different input here would not make sense after overwriting the sub register in the instruction before. I mean I might happen but it would not make sense.

Reply:
I have just implemented the check just in case.

Reply:
It may not make sense, but that doesn't mean that it cannot happen. E.g. in obfuscated binaries you have to expect operations that are only there to confuse. So we should make sure that such corner cases do not lead to errors.